### DISCUSSION TITLE
How to make ALE completion window to appear upon a key-stroke?

### DISCUSSION THREAD

**Discussion by ubaldot** [w0rp/ale#4079] [Q&A - Ask for help with problems]
The more I learn about `ALE`, the more I like it! :)
Today, I have discovered this 

`let g:ale_completion_enabled = 1`

that works fantastic! 
However, I find quite annoying that the completion window pop-up while I am typing.
Would that be possible to make it to appear only upon a certain key combination/keystroke, for example by pressing `<tab> `(or better, by exploiting [supertab](https://github.com/ervandew/supertab)?) 

Many thanks! :)

**Comment by hsanson** [comment]
Reading the help on ale-completion:

> ALE provides an 'omnifunc' function ale#completion#OmniFunc for triggering
completion manually with CTRL-X CTRL-O. i_CTRL-X_CTRL-O
>
>  " Use ALE's function for omnicompletion.
>  set omnifunc=ale#completion#OmniFunc

Not tested but I think this means that you can unset g:ale_completion_enabled, set the omnifunc to ALE's OmniFunc and then use the default CTRL-X CTRL-O to invoke the auto-completion.

**Comment by sukima** [comment]
This does not work! [Issue has been posted](https://github.com/dense-analysis/ale/issues/4370)

**Comment by sukima** [comment]
I just tried again with the following (and latest version) and it worked. Thought it took a good 30 seconds to get the first result. After that things went smoothly. Perhaps there is a lazy loading of the tsserver (LSP) happening?

```
let g:ale_completion_enabled = 0
set omnifunc=ale#completion#OmniFunc
```

A tip is to use the (not documented) insert mode map they add for you <kbd>CTRL+X CTRL+O CTRL+P</kbd> which immediately runs the completion and opens the popup menu without filling in while after that <kbd>CTRL+X CTRL+O</kbd> seems to work as expected.
